src/plugins/robot-memory/event_trigger_manager.cpp

```diff
@@ -84,21 +85,24 @@ EventTriggerManager::check_events()
 	for (EventTrigger *trigger : triggers) {
 		bool ok = true;
 		try {
-			while (trigger->oplog_cursor->more()) {
-				BSONObj change = trigger->oplog_cursor->next();
+			auto next = trigger->oplog_cursor.begin();
+			while (next != trigger->oplog_cursor.end()) {
 				//logger_->log_info(name.c_str(), "Triggering: %s", change.toString().c_str());
 				//actually call the callback function
-				trigger->callback(change);
+				trigger->callback(*next);
+				next++;
 			}
-		} catch (mongo::DBException &e) {
+		} catch (operation_exception &e) {
 			logger_->log_error(name.c_str(), "Error while reading the oplog");
 			ok = false;
 		}
-		if (!ok || trigger->oplog_cursor->isDead()) {
+		// TODO Do we still need to check whether the cursor is dead?
```

Comment:
http://mongocxx.org/api/current/classmongocxx_1_1cursor.html#details reads:
> By default, cursors timeout after 10 minutes of inactivity.

Also: https://github.com/mongodb/mongo-cxx-driver/blob/23f3d1d3b0f2bdbb247683cdcfb0e77b221841ba/src/mongocxx/cursor.cpp#L70-L75

Reply:
Fixed in 42c9e53c4274d608302e94e2d0857d76236fa0cf